### PR TITLE
`yarn.config.cjs` support

### LINKED ISSUE #597: Detect `yarn.config.cjs`
The new JS-based constraints. Available in Yarn 4+. [Docs](https://yarnpkg.com/features/constraints). Currently results in:

```
Unused files

yarn.config.cjs

Unused devDependencies          
       
@yarnpkg/types      package.json
```

### PR DESCRIPTION
**Pull request by jwoo0122**
Close #597

### CODE REVIEW

File: packages/knip/test/plugins/yarn.test.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+const cwd = resolve('fixtures/plugins/yarn');
+
+test('Find dependencies with the yarn plugin', async () => {
+  const { issues, counters } = await main({
+    ...baseArguments,
+    cwd,
+  });
+
+  assert(issues.files.size === 0);
```
Comment (webpro) [inline packages/knip/test/plugins/yarn.test.ts:16]: Just saying, `baseCounters` contains `files: 0` which is essentially the same assertion (no need to change though).

File: packages/knip/src/plugins/yarn/index.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import type { IsPluginEnabled, Plugin } from '../../types/config.js';
+import { _glob } from '../../util/glob.js';
+
+// https://yarnpkg.com/features/constraints
+
+const title = 'yarn';
```
Comment (webpro) [inline packages/knip/src/plugins/yarn/index.ts:6]: The only nitpick I have is that Yarn seems to be the official spelling (title cased).

File: packages/knip/src/plugins/yarn/index.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import type { IsPluginEnabled, Plugin } from '../../types/config.js';
+import { _glob } from '../../util/glob.js';
+
+// https://yarnpkg.com/features/constraints
+
+const title = 'Yarn';
+
+const enablers = 'This plugin is enabled when a `yarn.lock` file is found in the root folder.';
+
+const isEnabled: IsPluginEnabled = async ({ cwd }) => (await _glob({ cwd, patterns: ['yarn.lock'] })).length > 0;
```
Comment (webpro) [inline packages/knip/src/plugins/yarn/index.ts:10]: There's `_firstGlob` for this, any chance we could use that? Example: https://github.com/webpro-nl/knip/blob/main/packages/knip/src/plugins/github-actions/index.ts#L13-L14 (probably should move into a nice helper but we could consider this in another PR)

Comment (jwoo0122) [reply packages/knip/src/plugins/yarn/index.ts]: Great! I've changed it 0b8a898

**Comment by webpro** [comment]
Thanks, Jeong!

**Comment by webpro** [comment]
:rocket: _This pull request is included in v5.39.0. See [Release 5.39.0](https://github.com/webpro-nl/knip/releases/tag/5.39.0) for release notes._

_Using Knip in a commercial project? Please consider [becoming a sponsor](https://knip.dev/sponsors)._

**Pull request merged**